.github/workflows/ci.yaml

```
@@ -0,0 +1,60 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+env:
+  DEFAULT_GO_VERSION: "~1.23.0"
+jobs:
+  generate-and-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.DEFAULT_GO_VERSION }}
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Generate
+        run: make clean protobuf
+      - name: Check Clean Work Tree
+        run: make check-clean-work-tree
+  compatibility-test:
+    strategy:
+      matrix:
+        go-version: ["1.23.0", "~1.22.4"]
```

I think we should use the latest versions (as these are supported by the Go team)

```suggestion
        go-version: ["~1.23.0", "~1.22.4"]
```